### DISCUSSION TITLE
MQTT Sensor value in Fluidd interface

### DISCUSSION THREAD

**Discussion by midnight1979** [fluidd-core/fluidd#1333] [General]
Hello!
is it posible to show MQTT sensor value that is stored in moonracker somewhere in Fluidd interface????
and if it is not, does it possible to add this feature in fluidd??

**Comment by pedrolamas** [comment]
Assuming you are referring to Moonraker's [[sensor]](https://moonraker.readthedocs.io/en/latest/configuration/#sensor), we already support these in Fluidd since https://github.com/fluidd-core/fluidd/releases/tag/v1.26.0

You can see a screenshot of it on #1196

**Reply by midnight1979** [reply]
Oh, yes now I see, thanks! does it possible to add this temperature sensor in chart? 
And also how i can read sensor values in macros, I tryed to find some information about this, but couldn't find anything...

**Reply by pedrolamas** [reply]
I don't think it is possible to use these values in Klipper macros... these sensors are exposed directly by Moonraker so Klipper doesn't even know they exist.

As for the history, I can't recall if I did confirm that Moonraker was storing the values (it's been a while since I implemented this feature), but the problem here is that these values are considered just plain values - could be power, energy, length, etc., so there is nothing telling Fluidd that a specific value is a temperature!

**Reply by midnight1979** [reply]
I'm not sure, but it possible to use remote action_call_remote_method "server.sensors.measurements" in macros with junja2 statements, but there is no any information about sintax how to get the last sensor value for example.

**Reply by pedrolamas** [reply]
I believe that only works in the opposite way, so it is Klipper sending data to Moonraker!

**Reply by midnight1979** [reply]
but you did it in fluidd :) maybe i can get this value in macros by sensor name for example????? macro with junja2 template maybe...

**Reply by pedrolamas** [reply]
Fluidd accesses the Moonraker websocket to get these values.

Currently, Klipper only pushes data to Moonraker, AFAIK there is no data going the other way around and no way of doing it (unless something changes on Klipper side)

**Reply by midnight1979** [reply]
Very sad :(((((, some kind of "cutted integation"...

Thanks for your advice!
